### ISSUE TITLE
how to use Finetuned model

### ISSUE THREAD

**Issue by MJAnderson1** [SWivid/F5-TTS#339]
How do I use the finetuned model with gradio interface.  
The option to choose a model is only available in the train gradio ui

**Comment by jpgallegoar** [comment]
Right now, the only way to select a finetuned model is to change the code. I'm thinking of creating a Finetuning hub where you can upload different finetuned models for the community to use (and also integrate them in the gradio app, with local selection as well). What do you think @SWivid.

**Comment by rikabi89** [comment]
> Right now, the only way to select a finetuned model is to change the code. I'm thinking of creating a Finetuning hub where you can upload different finetuned models for the community to use (and also integrate them in the gradio app, with local selection as well). What do you think @SWivid.

That would be great. I mean you can tell the model on the "test model" in the finetuning gradio but its a bit limited. One thing I find annoying it asks for the reference text of the ref sample. I think this should be just transcribed auto.

**Comment by rikabi89** [comment]
![{2889FEAB-4DFA-495B-9626-A74EF99039FA}](https://github.com/user-attachments/assets/80962fc5-0dd5-43f6-9930-a1e10e564808)

**Comment by jpgallegoar** [comment]
Yes, I'm using that right now. It's not supposed to be comfortable, finetuning is not user friendly by definition. You can use [this](https://huggingface.co/spaces/hf-audio/whisper-large-v3-turbo) to transcribe if you want

**Comment by SWivid** [comment]
> Right now, the only way to select a finetuned model is to change the code. 

will only enable pretrained model for gradio interface as it's synchronized with huggingface space demo

> I'm thinking of creating a Finetuning hub where you can upload different finetuned models for the community to use (and also integrate them in the gradio app, with local selection as well).

it is easy for user to load local model for inference, as all the pipeline is there, just need to change the ckpt path.

basically the project is still an academic research, not ready and that mature for applications; 
we are probably focusing on the model itself

**Comment by jpgallegoar** [comment]
@SWivid I understand you are focusing on the model itself, but would you be opposed to me developing something like I said? For example, if I finetune a good spanish model, and some other person has a good japanese model, etc. Would you be ok with me creating an option to load these from the F5 app? Otherwise I can create a fork and do it there. Another option is to create it outside of infer_gradio.py but in this repo, in another gradio app. What option do you prefer?

**Comment by SWivid** [comment]
@jpgallegoar actually our point is, where the infer_gradio.py is to deploy

**Comment by SWivid** [comment]
> @SWivid Sorry, I don't understand what you mean. Do you mean this could affect HuggingFace spaces and such?

we know that huggingface has limited quota, and will it has some problems to load finetuned models and how the dropdown list are to defined (which models are ready for present?)
as for fish.audio they are ready for this kind of things as there're functions of registration, quota, user custom uploading, etc. but f5 is far from that.

and if locally use custom finetuned model, we could simply put a whole page of readme in `infer/` with clear instruction pointing to that, and people are welcome to share their hub with PR operation adding the link to there.

**Comment by SWivid** [comment]
@jpgallegoar ah, and i just recall that there exist way for local deployment with ckpt switching and will not influence online space demo.

just with https://github.com/SWivid/F5-TTS/blob/dee0420b59819cea85917d0bb15521aa96bf3f5b/src/f5_tts/infer/infer_gradio.py#L727

then we could simply have a input textbox e.g. filling with any ckpt hub path
Just an example, there would be better ways~

**Comment by jpgallegoar** [comment]
> @jpgallegoar ah, and i just recall that there exist way for local deployment with ckpt switching and will not influence online space demo.
> 
> just with
> 
> https://github.com/SWivid/F5-TTS/blob/dee0420b59819cea85917d0bb15521aa96bf3f5b/src/f5_tts/infer/infer_gradio.py#L727
> 
> then we could simply have a input textbox e.g. filling with any ckpt hub path Just an example, there would be better ways~

True! I will definitely look into this because there's a big spanish community looking for finetunes (and probably from other countries as well). I think it would be better to have it all in the same app for ease of use

**Comment by lpscr** [comment]
> Yes, I'm using that right now. It's not supposed to be comfortable, finetuning is not user friendly by definition. You can use [this](https://huggingface.co/spaces/hf-audio/whisper-large-v3-turbo) to transcribe if you want

from the gradio finetune 
 @rikabi89  you can load ref you need click x and load audio and if you use ref_text="" empty you get auto transcribe  , 
make sure you have last update

**Comment by rikabi89** [comment]
> ""

When I try this I just get a jibberish output which is a lot longer than what I wanted outputted

**Comment by lpscr** [comment]
@rikabi89 what language ?

**Comment by rikabi89** [comment]
> =""

EN

**Comment by lpscr** [comment]
@rikabi89 are you sure you have last update because with last update must woking 

https://github.com/SWivid/F5-TTS/blob/dee0420b59819cea85917d0bb15521aa96bf3f5b/src/f5_tts/api.py#L102

try the api also
      
```
    f5tts = F5TTS(ckpt_file="")

    wav, sr, spect = f5tts.infer(
        ref_file=str(files("f5_tts").joinpath("infer/examples/basic/basic_ref_en.wav")),
        ref_text="",
        gen_text="""I don't really care what you call me. I've been a silent spectator, watching species evolve, empires rise and fall. But always remember, I am mighty and enduring. Respect me and I'll nurture you; ignore me and you shall face the consequences.""",
        file_wave="output.wav",
        seed=-1,  # random seed = -1
    )
```

**Comment by jpgallegoar** [comment]
@lpscr Do you know why this is?
![image](https://github.com/user-attachments/assets/311b9663-be59-45d9-bd16-a17dd6180c59)
After restarting, loss looks really bad )

**Comment by rikabi89** [comment]
> @rikabi89 are you sure you have last update because with last update must working
> 
> https://github.com/SWivid/F5-TTS/blob/dee0420b59819cea85917d0bb15521aa96bf3f5b/src/f5_tts/api.py#L102
> 
> try the api also if working in api must working same in gradio finetune because use api
> 
> ```
>     f5tts = F5TTS(ckpt_file="")
> 
>     wav, sr, spect = f5tts.infer(
>         ref_file=str(files("f5_tts").joinpath("infer/examples/basic/basic_ref_en.wav")),
>         ref_text="",
>         gen_text="""I don't really care what you call me. I've been a silent spectator, watching species evolve, empires rise and fall. But always remember, I am mighty and enduring. Respect me and I'll nurture you; ignore me and you shall face the consequences.""",
>         file_wave="output.wav",
>         seed=-1,  # random seed = -1
>     )
> ```
> 
> to debug write in line 103 `print(ref_text)` this print the transcribe you get for the audio

![{2CADFDD4-DFC9-4539-9A82-5301A7CBDE45}](https://github.com/user-attachments/assets/99225e82-c1ed-4077-a20d-0dc3b59a33a7)
![{D2BA4560-0C76-4340-A21C-9C4E1573EF8B}](https://github.com/user-attachments/assets/2386f57b-521e-47f3-8c9c-76ff6ccafb33)

I've pulled latest update and still have this issue. Added print but i cant see any debug..

**Comment by rikabi89** [comment]
![{914DB6C9-1C0B-4C51-8632-3BBB0F142257}](https://github.com/user-attachments/assets/1bd2b0a4-9416-4aec-ba5f-5f302aec44fd)
Sorry ignore last screenshot I was in wrong folder...this is the one I am using. Will try the debug again

**Comment by rikabi89** [comment]
it working now, not sure why lol. Thanks

**Comment by SWivid** [comment]
will close this issue, move discussion to #396 .
